From 0a6ba0915b606dbb70e6d16fcf8be56edb63dedf Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <rstoyanov@fedoraproject.org>
Date: Fri, 4 Feb 2022 20:41:07 +0000
Subject: [PATCH 117/249] criu-ns: use os.waitstatus_to_exitcode()

os.WEXITSTATUS() returns the process exit status and it should be used
only if WIFEXITED() is true, i.e., the process terminated normally.

os.waitstatus_to_exitcode() does the same as os.WEXITSTATUS() but it
also handles the case when the process has been terminated by a signal.

Suggested-by: Andrei Vagin <avagin@gmail.com>
Signed-off-by: Radostin Stoyanov <rstoyanov@fedoraproject.org>
---
 scripts/criu-ns | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/criu-ns b/scripts/criu-ns
index 6aa618218..930d20c80 100755
--- a/scripts/criu-ns
+++ b/scripts/criu-ns
@@ -68,7 +68,7 @@ def _wait_for_process_status(criu_pid):
         try:
             (pid, status) = os.wait()
             if pid == criu_pid:
-                return os.WEXITSTATUS(status)
+                return os.waitstatus_to_exitcode(status)
         except OSError:
             return -251
 
-- 
2.35.1

